### DISCUSSION TITLE
Problems with Vision model

### DISCUSSION THREAD

**Discussion by VitoJanko** [567-labs/instructor#289] [Q&A] [answered]
Hello, 
Tried using this library and had good success with it using gpt-4-1106-preview model. However, when using  gpt-4-vision-preview, it crashes with the error message: 

`openai.BadRequestError: Error code: 400 - {'error': {'message': '2 validation errors for Request\nbody -> function_call\n  extra fields not permitted (type=value_error.extra)\nbody -> functions\n  extra fields not permitted (type=value_error.extra)', 'type': 'invalid_request_error', 'param': None, 'code': None}}
`

Is there some workaround for that? I might be missing something obvious

**Comment by jxnl** [comment]
Please share your code snippet

**Comment by jxnl** [comment, marked as answer]
https://jxnl.github.io/instructor/concepts/patching/?h=#markdown-json-mode

make sure you use the right now as function calling is not supported for vision models

**Reply by VitoJanko** [reply]
This seems to work, thanks!
